ArduPlane/Attitude.cpp

```diff
@@ -377,7 +377,10 @@ void Plane::stabilize()
     }
     float speed_scaler = get_speed_scaler();
 
-    if (quadplane.in_tailsitter_vtol_transition()) {
+    uint32_t now = AP_HAL::millis();
+    // time check deals with a race condition where this function is called before Quadplane has a chance to update the transition state upon entering a VTOL mode
+    const bool tailsitter_vtol_transition = quadplane.in_vtol_mode() && quadplane.is_tailsitter() && ((quadplane.transition_state == QuadPlane::TRANSITION_ANGLE_WAIT_VTOL) || ((now - quadplane.last_vtol_mode_ms) > 1000));
```

Comment:
it would be nice not to duplicate this logic, add a new tailsitter helper if needed, or add timestamp, with zero default

Reply:
added a timestamp